src/main/webapp/app/entities/file-upload-exercise/file-upload-exercise-detail.component.ts

```diff
@@ -20,15 +21,18 @@ export class FileUploadExerciseDetailComponent implements OnInit, OnDestroy {
 
     ngOnInit() {
         this.subscription = this.route.params.subscribe(params => {
-            this.load(params['id']);
+            this.load(params['exerciseId']);
         });
         this.registerChangeInFileUploadExercises();
     }
 
-    load(id: number) {
-        this.fileUploadExerciseService.find(id).subscribe((fileUploadExerciseResponse: HttpResponse<FileUploadExercise>) => {
-            this.fileUploadExercise = fileUploadExerciseResponse.body!;
-        });
+    load(exerciseId: number) {
+        this.fileUploadExerciseService
+            .find(exerciseId)
+            .pipe(filter(res => !!res.body))
+            .subscribe((fileUploadExerciseResponse: HttpResponse<FileUploadExercise>) => {
```

Comment:
It would be nice to have some error handling here.

Reply:
Good suggestion. Added.